cpp/src/arrow/ipc/read_write_test.cc

```diff
@@ -409,6 +414,32 @@ class IpcTestFixture : public io::MemoryMapFixture, public ExtensionTypesMixin {
     CheckRoundtrip(*batch, options, IpcReadOptions::Defaults(), buffer_size);
   }
 
+  void CheckRoundTrip(const RecordBatch& src_batch,
+                      const RecordBatch& expected_batch,
+                      IpcWriteOptions options = IpcWriteOptions::Defaults(),
+                      IpcReadOptions read_options = IpcReadOptions::Defaults(),
+                      int64_t buffer_size = 1 << 20) {
+    std::stringstream ss;
+    ss << "test-write-row-batch-" << g_file_number++;
+    ASSERT_OK_AND_ASSIGN(mmap_,
+                         io::MemoryMapFixture::InitMemoryMap(buffer_size, ss.str()));
+
+    DictionaryMemo dictionary_memo;
+
+    std::shared_ptr<Schema> schema_result;
+    DoSchemaRoundTrip(*src_batch.schema(), &dictionary_memo, &schema_result);
+    ASSERT_TRUE(expected_batch.schema()->Equals(*schema_result));
```

Comment:
Should we always convert schema's endianness to native endian?
I think that we should keep the original schema's endianness for serialization.
Did we discuss this?

Reply:
Thank you for your comment. You are right. We should keep the original schema until `ReadRecordBatch`.   
Then, if `LoadRecordBatchSubset` will automatically convert the endian based on the schema's endianness, `LoadRecordBatchSubset` will update the endian flag in the schema.

I will move this assertion after `Do*RoundTrip`.